pkg/cloud_provider/file/file.go

```diff
@@ -591,13 +590,13 @@ func IsUserError(err error) bool {
 	if !ok {
 		return false
 	}
-	userErrors := map[code.Code]bool{
-		code.Code_RESOURCE_EXHAUSTED: true,
-		code.Code_PERMISSION_DENIED:  true,
-		code.Code_INVALID_ARGUMENT:   true,
-		code.Code_NOT_FOUND:          true,
+	userErrors := map[int]bool{
+		http.StatusTooManyRequests: true,
```

Comment:
have you tried this change out manually?

Reply:
would it make sense to check on both http error code and grpc error code? based on response from Filestore API there's both info [available](https://screenshot.googleplex.com/C5Wue5uNf9JukV8).

Reply:
I tried it out for everything except filestore api not enabled because it won't let me disable mine on my current project because I have resources still using it. From the error codes I am filtering out, Http error 400 is the only error that has multiple grpc codes, and all the http 400 error codes are user errors so we could do that but it isn't necessary

Reply:
make sense

Reply:
you can delete any existing filestore resources in the project and try out `gcloud services disable  "file.googleapis.com" --force`

Reply:
Also thinking a bit I think 429 should be treated as user error. Let's discuss this a bit more offline. We ideally want to be alerted on quota exceed errors.

Reply:
I tried force as well and it still won't let me 
`ERROR: (gcloud.services.disable) The operation "operations/acf.p17-876590250267-9939fe22-4d40-49ec-83c4-fc57675d9e58" resulted in a failure "[Error in service 'file.googleapis.com': Not ready to deactivate the service on this project; ensure there are no more resources managed by this service.`